src/operations.jl

```diff
@@ -4,25 +4,75 @@ hash(c::TransparentGray, hx::UInt) = hash(alpha(c), hash(gray(c), hx))
 hash(c::AbstractRGB, hx::UInt) = hash(blue(c), hash(green(c), hash(red(c), hx)))
 hash(c::TransparentRGB, hx::UInt) = hash(alpha(c), hash(blue(c), hash(green(c), hash(red(c), hx))))
 
+# range
+range{T<:HSV}(::Type{T}) = (360,1,1)
+range{T<:HSL}(::Type{T}) = (360,1,1)
+range{T<:Lab}(::Type{T}) = (100,1,1)
+range{T<:LCHab}(::Type{T}) = (100,1,360)
+typealias Colorant1{T} Colorant{T,1}
+typealias Colorant2{T} Colorant{T,2}
+typealias Colorant3{T} Colorant{T,3}
+typealias Colorant4{T} Colorant{T,4}
+range{T<:Colorant1}(::Type{T}) = (1,)
```

Comment:
Nope, I don't really know what it's about. My main strategy would simply be to recommend *against* providing a generic fallback. Only have the method work for color types you want to support. Previously we only supported gray and RGB because those were the ones I cared about. This is much nicer, of course, but no need to kill yourself over types that people may not ever use `rand` with. After all, if they want random `DIN` colors, they can always say `convert(DIN99d, rand(RGB{Float32}))` or whatever.